### PR TITLE
feat(checkout, clone, push): Add 'onPrePush' and 'onPostCheckout'

### LINKED ISSUE #1186: Add support for Git Hooks
I'd love to see support for the git hooks. Some of these hooks are:

- [ ] `pre-commit`
- [ ] `prepare-commit-msg`
- [ ] `commit-msg`
- [ ] `post-commit`
- [ ] `post-merge`
- [ ] `pre-push`

And others. They can be found via [this resource](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) and [this resource](digitalocean.com/community/tutorials/how-to-use-git-hooks-to-automate-development-and-deployment-tasks) in a more "plain-english" way.

I'm willing to implement this functionality, however, I'm wondering what the API should look like. Do we want to look for the related files and run them with `spawn`? Do we want to make `pre-X` and `post-X` actions and [utilize a plugin-like system](https://github.com/isomorphic-git/isomorphic-git/issues/218) and provide this logic as an additional package? What do we think?

### PR DESCRIPTION
**Pull request by CoalZombik**
## I'm adding a parameter to an existing command X:

- [x] add parameter to the function in `src/api/X.js` (and `src/commands/X.js` if necessary)
- [x] document the parameter in the JSDoc comment above the function
- [x] add a test case in `__tests__/test-X.js` if possible
- [x] if this is your first time contributing, run `npm run add-contributor` and follow the prompts to add yourself to the README
- [x] squash merge the PR with commit message "feat(X): Added 'bar' parameter"
- [x] add page to the Docs Sidebar `website/sidebars.json`

Closes #1186

I added new documentation `website/sidebars.json`, but I don't know how to handle versioned variants of documentation.

### CODE REVIEW

File: docs/onPostCheckout.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+For more information, see [`post-checkout` git documentation](https://git-scm.com/docs/githooks#_post_checkout).
+
+## Example
+
+```js
+await git.checkout({
+  ...,
+  onPrePush: args => {
+    console.log(args)
+  }
```
Comment (jcubic) [inline docs/onPostCheckout.md:34]: Wrong hook in docs

File: docs/onPrePush.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+---
+title: onPrePush
+sidebar_label: onPrePush
+---
+
+The `onPrePush` callback is called before sending an object pack to remote and can be used to abort the push action.
+
+This callback is implemented as an equivalent to the canonical git `pre-push` hook. An `onPrePush` function is passed an object containing information about the target remote and `url`, local and remote `ref` name and the commit oids. This function must return `true` if the push action is to be aborted and `false` if not.
```
Comment (jcubic) [inline docs/onPrePush.md:8]: I would reverse the values. I know that on Linux exit code 0 is all fine, but in JavaScript true is all fine, and false is that something is wrong. So this is not very intuitive.

File: src/commands/push.js [hunk 3]
```diff
@@ -137,6 +140,16 @@ export async function _push({
...
     '0000000000000000000000000000000000000000'
 
+  if (onPrePush) {
+    const hookCancel = await onPrePush({
+      remote,
+      url,
+      localRef: { ref: _delete ? '(delete)' : fullRef, oid: oid },
+      remoteRef: { ref: fullRemoteRef, oid: oldoid },
+    })
+    if (hookCancel) throw new UserCanceledError()
```
Comment (jcubic) [inline src/commands/push.js:150]: Reverse the condition

**Review by jcubic — CHANGES_REQUESTED** [review]
Few comments

**Review by jcubic — CHANGES_REQUESTED** [review]
Few comments

**Comment by jcubic** [comment]
Sorry for late reply. Did you change something? I didn't get any notification from GitHub. And you didn't reply to my comments.

**Comment by CoalZombik** [comment]
Yes, I applied changes that you suggested.

Sorry, it probably my fault, I expected that re-requesting a review will send you a notification.

**Comment by jcubic** [comment]
Hey, sorry for late response, I totally forget about this Review. I've found my old notes where I did add a link to the PR.

Looks good.

**Review by jcubic — COMMENTED** [review]
Looks good

**Pull request merged**